core/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -160,23 +160,80 @@ class OptimizeExecutor(
 
   private val isMultiDimClustering = zOrderByColumns.nonEmpty
 
-  def optimize(): Seq[Row] = {
+  def optimize(isAutoCompact: Boolean = false, targetFiles: Seq[AddFile] = Nil): Seq[Row] = {
```

Comment:
`isAutoCompact` caught my attention (after `auto` above in `Optimize`) but don't really know how to make these two names the same. I feel they should really be the same but no idea how. Sorry.

Reply:
That `Optimize` is for deltaLog. The current name looks fine to me but let me know a better one if any.